src/aws-cpp-sdk-core/source/utils/logging/DefaultLogSystem.cpp

```diff
@@ -90,12 +96,27 @@ DefaultLogSystem::~DefaultLogSystem()
         m_syncData.m_queueSignal.notify_one();
     }
 
+    // explicitly wait for logging thread to finish
+    {
+        std::unique_lock<std::mutex> locker(m_syncData.m_logQueueMutex);
+        if (!m_syncData.m_loggingThreadStopped)
```

Comment:
Should this be "while" instead of "if" and not use a conditional wait_for?  It is not safe to leave the destructor until we know the thread is gone.

Reply:
there is .join for the thread later.
I just did not want to block user code if the boolean is never set.